In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns


In [2]:
#import os
#current_dir = os.getcwd()
#os.getcwd() test

## Import dataset
<p>Premièrement, on va importer les fichiers CSV des entreprises, de la géographie, des salaires et de la population et afficher l’aperçu.<p>


In [42]:
# Import datasets with DropBox

#df = pd.read_csv("https://www.dropbox.com/s/asdfasdfasfasdf/movie_data.csv?dl=1")
population_url = "https://www.dropbox.com/scl/fi/dh41m7t7349tuephx50i0/population.csv?rlkey=b66y292vmbge3zk6y7png49o6&dl=1"
salary_url ="https://www.dropbox.com/scl/fi/ni4mfl21r4dv2eipdqpet/net_salary_per_town_categories_update2021.csv?rlkey=8aodl7dbfrwvwbz397gzkht6o&dl=1" 
name_geographic_url = "https://www.dropbox.com/scl/fi/zmeoiz9y21hho7zin0nr6/name_geographic_final.csv?rlkey=uvjffxfzozrzaia6i39xpb3mf&dl=1"
enterprise_url = "https://www.dropbox.com/scl/fi/7hf3w5emqxu11kqpyayr7/df_entreprises24.csv?rlkey=52jgbm2kg1o3x1ab8gsf689ht&dl=1"

df_population = pd.read_csv(population_url, dtype = {'CODGEO': object})
print("----------------POPULATION----------------")
display(df_population)

df_salary = pd.read_csv(salary_url, dtype = {'CODGEO': object}, sep = ';')
print("-----------------SALARY-----------------")
display(df_salary)

df_enterprise = pd.read_csv(enterprise_url,dtype = {'CODGEO': object}, sep = ',')
print("----------------ENTERPRISE----------------")
display(df_enterprise)

df_name_geographic = pd.read_csv(name_geographic_url,  sep = ',')
print("----------------NAME-GEOGRAPHIC----------------")
display(df_name_geographic)


----------------POPULATION----------------


NIVGEO CODGEO                   LIBGEO  MOCO  AGEQ80_17  SEXE  NB
0          COM  01001  L'Abergement-Clémenciat    11          0     1  15
1          COM  01001  L'Abergement-Clémenciat    11          0     2  15
2          COM  01001  L'Abergement-Clémenciat    11          5     1  20
3          COM  01001  L'Abergement-Clémenciat    11          5     2  20
4          COM  01001  L'Abergement-Clémenciat    11         10     1  20
...        ...    ...                      ...   ...        ...   ...  ..
8536579    COM  97424                   Cilaos    32         70     2  34
8536580    COM  97424                   Cilaos    32         75     1   8
8536581    COM  97424                   Cilaos    32         75     2  42
8536582    COM  97424                   Cilaos    32         80     1   0
8536583    COM  97424                   Cilaos    32         80     2  21

[8536584 rows x 7 columns]

-----------------SALARY-----------------


CODGEO     SNHM20    SNHMC20    SNHMP20    SNHME20    SNHMO20    SNHMF20  \
0      1004  15.013132  25.221939  16.950395  11.259209  11.772666  13.043695   
1      1007  15.261333  24.391671  16.179052  11.917058  12.520221  13.855898   
2      1014  14.578709  27.940065  16.384056  12.175141  11.848787  12.570863   
3      1024  14.658174  23.817275  16.319296  11.905547  12.127913  13.053780   
4      1025  14.952910  25.822076  15.365464  11.663794  12.400297  13.444614   
...     ...        ...        ...        ...        ...        ...        ...   
5416  75116  42.225452  56.361091  24.346292  13.612314  12.428167  30.951219   
5417  75117  30.828835  40.230738  19.159600  12.993530  11.611300  26.587149   
5418  75118  21.592317  29.996904  17.288963  12.239401  11.010065  21.043298   
5419  75119  18.573391  27.672799  16.707957  11.865190  10.769154  18.409916   
5420  75120  19.430558  27.412529  16.885757  12.315557  11.122145  19.003974   

       SNHMFC20   SNHMFP20   SNHMFE20  ...   SNHMHO20   SNHM1820   SNHM2620  \
0     21.806580  14.368130  11.045518  ...  12.095672  10.880315  15.072711   
1     21.532587  15.160246  11.951341  ...  12.881610  10.544073  15.263487   
2     20.702968  13.198429  12.152977  ...  12.304387  11.058325  14.116861   
3     20.157674  14.923592  11.414435  ...  12.525517  10.498073  14.627982   
4     22.271103  14.233353  11.410936  ...  12.666390  10.385396  14.750372   
...         ...        ...        ...  ...        ...        ...        ...   
5416  42.560258  19.867132  13.048285  ...  12.568153  14.428286  38.687379   
5417  34.087607  18.274744  12.863740  ...  11.717475  13.618584  30.001531   
5418  27.401397  17.121611  12.626235  ...  11.051186  12.272080  21.999066   
5419  25.828373  16.493335  12.134126  ...  10.838728  11.476194  18.890023   
5420  25.526068  16.684395  12.523575  ...  11.202774  11.787338  19.613933   

       SNHM5020  SNHMF1820  SNHMF2620  SNHMF5020  SNHMH1820  SNHMH2620  \
0     17.255747  10.291297  13.234401  14.192028  11.286232  16.130620   
1     16.961952  10.618428  14.034526  14.418987  10.487221  16.229187   
2     16.528222  10.295307  12.226430  13.932078  11.623660  15.517659   
3     16.404495  10.031113  13.268833  13.691905  10.812016  15.505663   
4     16.854860  10.309374  13.513608  14.336143  10.442608  15.580433   
...         ...        ...        ...        ...        ...        ...   
5416  55.532210  13.166830  30.919309  35.518382  15.806210  45.823466   
5417  39.670250  13.324798  26.964962  30.336282  13.944836  32.773196   
5418  23.595562  12.358482  21.722465  22.233565  12.184851  22.213960   
5419  20.024059  11.304342  18.934548  19.580112  11.630749  18.859667   
5420  21.336076  11.785832  19.247516  20.770531  11.788720  19.904173   

      SNHMH5020  
0     19.226540  
1     19.324302  
2     18.409024  
3     18.422798  
4     18.421533  
...         ...  
5416  73.486222  
5417  48.729793  
5418  24.657557  
5419  20.353672  
5420  21.792705  

[5421 rows x 25 columns]

----------------ENTERPRISE----------------


CODGEO  ET_BE  ET_BE_0sal  ET_BE_1_4  ET_BE_5_9  ET_BE_10_19  \
0      01001      0           0          0          0            0   
1      01002      0           0          0          0            0   
2      01004     33           2          8          6           12   
3      01005      4           0          1          1            2   
4      01006      0           0          0          0            0   
...      ...    ...         ...        ...        ...          ...   
34965  97420     40           4         17         11            5   
34966  97421     11           1          6          3            1   
34967  97422    135          11         67         29           22   
34968  97423      8           0          4          4            0   
34969  97424      7           0          4          2            1   

       ET_BE_20_49  ET_BE_50_99  ET_BE_100_199  ET_BE_200_499  ET_BE_500P  ||  \
0                0            0              0              0           0  ||   
1                0            0              0              0           0  ||   
2                3            2              0              0           0  ||   
3                0            0              0              0           0  ||   
4                0            0              0              0           0  ||   
...            ...          ...            ...            ...         ...  ..   
34965            3            0              0              0           0  ||   
34966            0            0              0              0           0  ||   
34967            5            1              0              0           0  ||   
34968            0            0              0              0           0  ||   
34969            0            0              0              0           0  ||   

       Total_Salaries  nb_auto_entrepreneur  nb_micro_entreprises  \
0                   0                     0                     0   
1                   0                     0                     0   
2                  33                     2                    14   
3                   4                     0                     2   
4                   0                     0                     0   
...               ...                   ...                   ...   
34965              40                     4                    28   
34966              11                     1                     9   
34967             135                    11                    96   
34968               8                     0                     8   
34969               7                     0                     6   

       nb_small_entreprises  nb_medium_entreprises  nb_large_entreprises  
0                         0                      0                     0  
1                         0                      0                     0  
2                        15                      2                     0  
3                         2                      0                     0  
4                         0                      0                     0  
...                     ...                    ...                   ...  
34965                     8                      0                     0  
34966                     1                      0                     0  
34967                    27                      1                     0  
34968                     0                      0                     0  
34969                     1                      0                     0  

[34970 rows x 18 columns]

/var/folders/m4/rl8q4_8s3531vrprrghc32pm0000gn/T/ipykernel_35164/2021177980.py:19: DtypeWarning: Columns (0,5,7,14) have mixed types. Specify dtype option on import or set low_memory=False.
  df_name_geographic = pd.read_csv(name_geographic_url,  sep = ',')


----------------NAME-GEOGRAPHIC----------------


COM_code_insee                  COM_nom  COM_latitude  COM_longitude  \
0                1001  L'Abergement-Clémenciat     46.153721       4.925850   
1                1001  L'Abergement-Clémenciat     46.153721       4.925850   
2                1002    L'Abergement-de-Varey     46.009606       5.428088   
3                1002    L'Abergement-de-Varey     46.009606       5.428088   
4                1004        Ambérieu-en-Bugey     45.961049       5.372275   
...               ...                      ...           ...            ...   
271872          97617                 Tsingoni    -12.782686      45.134576   
271873          97617                 Tsingoni    -12.782686      45.134576   
271874          97617                 Tsingoni    -12.782686      45.134576   
271875          97617                 Tsingoni    -12.782686      45.134576   
271876          97617                 Tsingoni    -12.782686      45.134576   

       DEPT DEPT_code DEPT_nom DEPT_ChefLieu_Code_insee    DEPT_ChefLieu  \
0        ||         1      Ain                     1053  bourg en bresse   
1        ||         1      Ain                     1053  bourg en bresse   
2        ||         1      Ain                     1053  bourg en bresse   
3        ||         1      Ain                     1053  bourg en bresse   
4        ||         1      Ain                     1053  bourg en bresse   
...     ...       ...      ...                      ...              ...   
271872   ||       976  Mayotte                    97611        mamoudzou   
271873   ||       976  Mayotte                    97611        mamoudzou   
271874   ||       976  Mayotte                    97611        mamoudzou   
271875   ||       976  Mayotte                    97611        mamoudzou   
271876   ||       976  Mayotte                    97611        mamoudzou   

        DEPT_ChefLieu_latitude  ...  REG_ChefLieu_latitude  \
0                    46.205014  ...              45.770061   
1                    46.205014  ...              45.770061   
2                    46.205014  ...              45.770061   
3                    46.205014  ...              45.770061   
4                    46.205014  ...              45.770061   
...                        ...  ...                    ...   
271872              -12.790232  ...             -12.790232   
271873              -12.790232  ...             -12.790232   
271874              -12.790232  ...             -12.790232   
271875              -12.790232  ...             -12.790232   
271876              -12.790232  ...             -12.790232   

       REG_ChefLieu_longitude  CAP Capitale Capitale_latitude  \
0                    4.828519   ||    Paris         48.852937   
1                    4.828519   ||    Paris         48.852937   
2                    4.828519   ||    Paris         48.852937   
3                    4.828519   ||    Paris         48.852937   
4                    4.828519   ||    Paris         48.852937   
...                       ...  ...      ...               ...   
271872              45.194781   ||    Paris         48.852937   
271873              45.194781   ||    Paris         48.852937   
271874              45.194781   ||    Paris         48.852937   
271875              45.194781   ||    Paris         48.852937   
271876              45.194781   ||    Paris         48.852937   

       Capitale_longitude  DIST  DIST_COM_CL_DEPT DIST_COM_CL_REG  \
0                 2.35005    ||             25.27           43.32   
1                 2.35005    ||             25.27           43.32   
2                 2.35005    ||             25.88           53.50   
3                 2.35005    ||             25.88           53.50   
4                 2.35005    ||             28.83           47.15   
...                   ...   ...               ...             ...   
271872            2.35005    ||              6.58            6.58   
271873            2.35005    ||              6.58            6.58  

In [26]:
## Problem with reading population, probably because it's too big 

# File ID of the CSV file
file_id_entreprise = '1_S5SqqWz0WmcHAHwJzEKNnRRrM7dLQ5x'
file_id_salary = '1NLw8ymnnzLONUM1IVrYsH_A7DsqT84df'
file_id_name_geographic = '1ROnTSTBlJ6LBmm6tz6luwcTR12U3___p'
file_id_population = '1LsrModG7yNKY_UI75UAXyb_4ixeUUxJj'

# Construct the direct download link
download_link_entreprise = f'https://drive.google.com/uc?export=download&id={file_id_entreprise}'
download_link_salary = f'https://drive.google.com/uc?export=download&id={file_id_salary}'
download_link_name_geographic = f'https://drive.google.com/uc?export=download&id={file_id_name_geographic}'
download_link_population = f'https://drive.google.com/uc?export=download&confirm=1&id={file_id_population}'

# Read the CSV file into a DataFrame
df_entreprise = pd.read_csv(download_link_entreprise)
display(df_entreprise.head())

df_salary = pd.read_csv(download_link_salary, sep = ';')
display(df_salary.head())

df_name_geographic = pd.read_csv(download_link_name_geographic)
display(df_name_geographic.head())

df_population = pd.read_csv(download_link_population)
display(df_population.head())

CODGEO                   LIBGEO  REG DEP  E14TST  E14TS0ND  E14TS1  E14TS6  \
0  01001  L'Abergement-Clémenciat   82  01      25        22       1       2   
1  01002    L'Abergement-de-Varey   82  01      10         9       1       0   
2  01004        Ambérieu-en-Bugey   82  01     996       577     272      63   
3  01005      Ambérieux-en-Dombes   82  01      99        73      20       3   
4  01006                  Ambléon   82  01       4         4       0       0   

   E14TS10  E14TS20  E14TS50  E14TS100  E14TS200  E14TS500  
0        0        0        0         0         0         0  
1        0        0        0         0         0         0  
2       46       24        9         3         2         0  
3        1        2        0         0         0         0  
4        0        0        0         0         0         0

CODGEO     SNHM20    SNHMC20    SNHMP20    SNHME20    SNHMO20    SNHMF20  \
0   1004  15.013132  25.221939  16.950395  11.259209  11.772666  13.043695   
1   1007  15.261333  24.391671  16.179052  11.917058  12.520221  13.855898   
2   1014  14.578709  27.940065  16.384056  12.175141  11.848787  12.570863   
3   1024  14.658174  23.817275  16.319296  11.905547  12.127913  13.053780   
4   1025  14.952910  25.822076  15.365464  11.663794  12.400297  13.444614   

    SNHMFC20   SNHMFP20   SNHMFE20  ...   SNHMHO20   SNHM1820   SNHM2620  \
0  21.806580  14.368130  11.045518  ...  12.095672  10.880315  15.072711   
1  21.532587  15.160246  11.951341  ...  12.881610  10.544073  15.263487   
2  20.702968  13.198429  12.152977  ...  12.304387  11.058325  14.116861   
3  20.157674  14.923592  11.414435  ...  12.525517  10.498073  14.627982   
4  22.271103  14.233353  11.410936  ...  12.666390  10.385396  14.750372   

    SNHM5020  SNHMF1820  SNHMF2620  SNHMF5020  SNHMH1820  SNHMH2620  SNHMH5020  
0  17.255747  10.291297  13.234401  14.192028  11.286232  16.130620  19.226540  
1  16.961952  10.618428  14.034526  14.418987  10.487221  16.229187  19.324302  
2  16.528222  10.295307  12.226430  13.932078  11.623660  15.517659  18.409024  
3  16.404495  10.031113  13.268833  13.691905  10.812016  15.505663  18.422798  
4  16.854860  10.309374  13.513608  14.336143  10.442608  15.580433  18.421533  

[5 rows x 25 columns]

EU_circo  code_région   nom_région chef.lieu_région numéro_département  \
0  Sud-Est           82  Rhône-Alpes             Lyon                 01   
1  Sud-Est           82  Rhône-Alpes             Lyon                 01   
2  Sud-Est           82  Rhône-Alpes             Lyon                 01   
3  Sud-Est           82  Rhône-Alpes             Lyon                 01   
4  Sud-Est           82  Rhône-Alpes             Lyon                 01   

  nom_département       préfecture  numéro_circonscription  \
0             Ain  Bourg-en-Bresse                       1   
1             Ain  Bourg-en-Bresse                       1   
2             Ain  Bourg-en-Bresse                       1   
3             Ain  Bourg-en-Bresse                       1   
4             Ain  Bourg-en-Bresse                       1   

            nom_commune codes_postaux  code_insee   latitude longitude  \
0              Attignat         01340        1024  46.283333  5.166667   
1              Beaupont         01270        1029  46.400000  5.266667   
2                  Bény         01370        1038  46.333333  5.283333   
3             Béreyziat         01340        1040  46.366667      5.05   
4  Bohas-Meyriat-Rignat         01250        1245  46.133333       5.4   

   éloignement  
0         1.21  
1         1.91  
2         1.51  
3         1.71  
4         1.01

Empty DataFrame
Columns: [<!DOCTYPE html><html><head><title>Google Drive - Virus scan warning</title><meta http-equiv="content-type" content="text/html; charset=utf-8"/><style nonce="HhTMoEfqxqeUrzDLyUNExw">.goog-link-button{position:relative;color:#15c;text-decoration:underline;cursor:pointer}.goog-link-button-disabled{color:#ccc;text-decoration:none;cursor:default}body{color:#222;font:normal 13px/1.4 arial, sans-serif;margin:0}.grecaptcha-badge{visibility:hidden}.uc-main{padding-top:50px;text-align:center}#uc-dl-icon{display:inline-block;margin-top:16px;padding-right:1em;vertical-align:top}#uc-text{display:inline-block;max-width:68ex;text-align:left}.uc-error-caption, .uc-warning-caption{color:#222;font-size:16px}#uc-download-link{text-decoration:none}.uc-name-size a{color:#15c;text-decoration:none}.uc-name-size a:visited{color:#61c;text-decoration:none}.uc-name-size a:active{color:#d14836;text-decoration:none}.uc-footer{color:#777;font-size:11px;padding-bottom:5ex;padding-top:5ex;text-align:center}.uc-footer a{color:#15c}.uc-footer a:visited{color:#61c}.uc-footer a:active{color:#d14836}.uc-footer-divider{color:#ccc;width:100%}.goog-inline-block{position:relative;display:-moz-inline-box;display:inline-block}* html .goog-inline-block{display:inline}*:first-child+html .goog-inline-block{display:inline}sentinel{}</style><link rel="icon" href="//ssl.gstatic.com/docs/doclist/images/drive_2022q3_32dp.png"/></head><body><div class="uc-main"><div id="uc-dl-icon" class="image-container"><div class="drive-sprite-aux-download-file"></div></div><div id="uc-text"><p class="uc-warning-caption">Google Drive can't scan this file for viruses.</p><p class="uc-warning-subcaption"><span class="uc-name-size"><a href="/open?id=1LsrModG7yNKY_UI75UAXyb_4ixeUUxJj">df_population.csv</a> (317M)</span> is too large for Google to scan for viruses. Would you still like to download this file?</p><form id="download-form" action="https://drive.usercontent.google.com/download" method="get"><input type="submit" id="uc-download-link" class="goog-inline-block jfk-button jfk-button-action" value="Download anyway"/><input type="hidden" name="id" value="1LsrModG7yNKY_UI75UAXyb_4ixeUUxJj"><input type="hidden" name="export" value="download"><input type="hidden" name="confirm" value="t"><input type="hidden" name="uuid" value="298b4625-218a-4e33-9bb1-e2ecb876ec3f"></form></div></div><div class="uc-footer"><hr class="uc-footer-divider"></div></body></html>]
Index: []

In [24]:
# Import and review the data 

df_entreprise = pd.read_csv('../data/base_etablissement_par_tranche_effectif.csv') 
display(df_entreprise.head())

df_salary = pd.read_csv('../data/net_salary_per_town_categories_update2021.csv', sep =';')
display(df_salary.head())

df_name_geographic = pd.read_csv('../data/name_geographic_information.csv') 
display(df_name_geographic.head())

df_population = pd.read_csv('../data/population.csv', dtype={'CODGEO': object} )
display(df_population.head())

CODGEO                   LIBGEO  REG DEP  E14TST  E14TS0ND  E14TS1  E14TS6  \
0  01001  L'Abergement-Clémenciat   82  01      25        22       1       2   
1  01002    L'Abergement-de-Varey   82  01      10         9       1       0   
2  01004        Ambérieu-en-Bugey   82  01     996       577     272      63   
3  01005      Ambérieux-en-Dombes   82  01      99        73      20       3   
4  01006                  Ambléon   82  01       4         4       0       0   

   E14TS10  E14TS20  E14TS50  E14TS100  E14TS200  E14TS500  
0        0        0        0         0         0         0  
1        0        0        0         0         0         0  
2       46       24        9         3         2         0  
3        1        2        0         0         0         0  
4        0        0        0         0         0         0

CODGEO     SNHM20    SNHMC20    SNHMP20    SNHME20    SNHMO20    SNHMF20  \
0   1004  15.013132  25.221939  16.950395  11.259209  11.772666  13.043695   
1   1007  15.261333  24.391671  16.179052  11.917058  12.520221  13.855898   
2   1014  14.578709  27.940065  16.384056  12.175141  11.848787  12.570863   
3   1024  14.658174  23.817275  16.319296  11.905547  12.127913  13.053780   
4   1025  14.952910  25.822076  15.365464  11.663794  12.400297  13.444614   

    SNHMFC20   SNHMFP20   SNHMFE20  ...   SNHMHO20   SNHM1820   SNHM2620  \
0  21.806580  14.368130  11.045518  ...  12.095672  10.880315  15.072711   
1  21.532587  15.160246  11.951341  ...  12.881610  10.544073  15.263487   
2  20.702968  13.198429  12.152977  ...  12.304387  11.058325  14.116861   
3  20.157674  14.923592  11.414435  ...  12.525517  10.498073  14.627982   
4  22.271103  14.233353  11.410936  ...  12.666390  10.385396  14.750372   

    SNHM5020  SNHMF1820  SNHMF2620  SNHMF5020  SNHMH1820  SNHMH2620  SNHMH5020  
0  17.255747  10.291297  13.234401  14.192028  11.286232  16.130620  19.226540  
1  16.961952  10.618428  14.034526  14.418987  10.487221  16.229187  19.324302  
2  16.528222  10.295307  12.226430  13.932078  11.623660  15.517659  18.409024  
3  16.404495  10.031113  13.268833  13.691905  10.812016  15.505663  18.422798  
4  16.854860  10.309374  13.513608  14.336143  10.442608  15.580433  18.421533  

[5 rows x 25 columns]

EU_circo  code_région   nom_région chef.lieu_région numéro_département  \
0  Sud-Est           82  Rhône-Alpes             Lyon                 01   
1  Sud-Est           82  Rhône-Alpes             Lyon                 01   
2  Sud-Est           82  Rhône-Alpes             Lyon                 01   
3  Sud-Est           82  Rhône-Alpes             Lyon                 01   
4  Sud-Est           82  Rhône-Alpes             Lyon                 01   

  nom_département       préfecture  numéro_circonscription  \
0             Ain  Bourg-en-Bresse                       1   
1             Ain  Bourg-en-Bresse                       1   
2             Ain  Bourg-en-Bresse                       1   
3             Ain  Bourg-en-Bresse                       1   
4             Ain  Bourg-en-Bresse                       1   

            nom_commune codes_postaux  code_insee   latitude longitude  \
0              Attignat         01340        1024  46.283333  5.166667   
1              Beaupont         01270        1029  46.400000  5.266667   
2                  Bény         01370        1038  46.333333  5.283333   
3             Béreyziat         01340        1040  46.366667      5.05   
4  Bohas-Meyriat-Rignat         01250        1245  46.133333       5.4   

   éloignement  
0         1.21  
1         1.91  
2         1.51  
3         1.71  
4         1.01

NIVGEO CODGEO                   LIBGEO  MOCO  AGEQ80_17  SEXE  NB
0    COM  01001  L'Abergement-Clémenciat    11          0     1  15
1    COM  01001  L'Abergement-Clémenciat    11          0     2  15
2    COM  01001  L'Abergement-Clémenciat    11          5     1  20
3    COM  01001  L'Abergement-Clémenciat    11          5     2  20
4    COM  01001  L'Abergement-Clémenciat    11         10     1  20

In [ ]:
# Review data types and columns info
print(df_entreprise.info())
print(df_salary.info())
print(df_name_geographic.info())
print(df_population.info())

In [ ]:
# Shape for each DataFrame 

print("Shape of DF:")
print("entreprise:",df_entreprise.shape)
print("salary:",df_salary.shape)
print("name_geographic:",df_name_geographic.shape)
print("population:",df_population.shape)


In [ ]:
# Number of duplicates for each DataFrame

print('Number of duplicates:')
print('df_entreprise:', df_entreprise.duplicated().sum())
print('salary:', df_salary.duplicated().sum())
print('name_geographic:', df_name_geographic.duplicated().sum())
print('population:', df_population.duplicated().sum())


In [ ]:
# Function that provides a short summary of DataFrame with information about the number of unique values, 
# percentage of missing values, number of missing values, and the data type of each column.

def summary_short(df):
    """
    Function that provides a short summary of DataFrame with information about the number of unique values, 
    percentage of missing values, number of missing values, and the data type of each column.
    
    Is used with not cleaned DataFrames.
    
    Parameters:
    df (DataFrame): The DataFrame for which the summary is required.
    
    Returns:
    DataFrame with 
    """
    
    tab = pd.DataFrame(index = df.columns, columns = ['nb_unique_values','%_missing_values','nb_missing_values','type'])
    tab.loc[:,'nb_unique_values'] = df.nunique().values
    tab.loc[:,'%_missing_values'] = df.isna().sum().values / len(df)
    tab.loc[:,'nb_missing_values'] = df.isna().sum().values
    tab.loc[:, 'type'] = df.dtypes.values
    
    return display(tab)

In [ ]:
summary_short(df_name_geographic)

In [ ]:
#df_population['CODGEO'].astype(float)

# Data type verification  
def detection_error(element):
    try:
        float(element)
    except ValueError:
        return element
    
print(df_name_geographic.longitude.apply(detection_error).loc[df_name_geographic.longitude.apply(detection_error).notna()].values)


## Conclusion pour name_geographic DF : 
- Les noms des colonnes en français, tandis que ceux des autres DataFrames sont en anglais. Il est donc nécessaire de les renommer pour une cohérence.
- La colonne "code_insee" semble correspondre à "CODGEO", mais il y a un '0' au debut, par example, '1024' au lieu de '01024'.
- Les colonnes "latitude", "longitude", "éloignement" présentent respectivement 2929, 2841 et 2962 valeurs manquantes (NaN).
- La colonne "longitude" devrait normalement être de type float64. Cependant, nous n'avons pas pu modifier le type en raison de problèmes avec les données. Par exemple :
  
    - L'utilisation de ',' au lieu de '.' pour la notation décimale : **'5,83'**
    - La présence de '-' : ceux-ci doivent être remplacés par des valeurs NaN.


In [ ]:
def summary(df):
    """
    Provides a summary of the DataFrame with information about the number of unique values, 
    percentage of missing values, data type of each column, mean or mode depending on the data type,
    and potential alerts such as too many missing values or imbalanced categories.
    
    Parameters:
    df (DataFrame): The DataFrame for which the summary is required.
    
    Returns:
    None
    """
    
    table = pd.DataFrame(
        index=df.columns,
        columns=['type_info', '%_missing_values', 'nb_unique_values', 'nb_zero_values','%_zero_values', 'list_unique_values', "mean_or_mode", "flag"])
    
    # Fill in column 'type_info' with data types of each column
    table.loc[:, 'type_info'] = df.dtypes.values
    
    # Calculate the percentage of missing values for each column
    table.loc[:, '%_missing_values'] = np.round((df.isna().sum().values / len(df)) * 100)
    
    # Calculate the number of unique values for each column
    table.loc[:, 'nb_unique_values'] = df.nunique().values

    # Calculate the number of 0 values for each column
    table.loc[:, 'nb_zero_values'] = df.isin([0]).sum().values

    # Calculate the percentage of 0 values for each column
    table.loc[:, '%_zero_values'] = np.round((df.isin([0]).sum().values/len(df)) * 100)
    
    # Function to get the list of unique values for a column, showing "Too many categories..." if there are too many
    def get_list_unique_values(colonne):
        if colonne.nunique() < 11:
            return colonne.unique()
        else:
            return "Too many categories..." if colonne.dtypes == "O" else "Too many values..."
    
    # Function to get the mean or mode depending on the data type
    def get_mean_mode(colonne):
        if (colonne.dtypes == "O") or (len(colonne.unique()) < 20):
            return colonne.astype('category').mode()[0]
        else: 
            return colonne.mean()
    
    # Function to check for potential alerts such as too many missing values or imbalanced categories
    def alerts(colonne, thresh_na=0.25, thresh_balance=0.8):
        if (colonne.isna().sum()/len(df)) > thresh_na:
            return "Too many missing values!"
        elif colonne.value_counts(normalize=True).values[0] > thresh_balance:
            return "It's imbalanced!"
        else:
            return "Nothing to report"
        
    # Fill in the columns with the respective information
    table.loc[:, 'list_unique_values'] = df.apply(get_list_unique_values)
    table.loc[:, 'mean_or_mode'] = df.apply(get_mean_mode)
    table.loc[:, 'flag'] = df.apply(alerts)
    
    # Display the summary table
    return display(table)

In [ ]:
summary(df_entreprise)

## Conclusion pour enterprise DF : 
- Il y a beaucoup de zeros dans la DF pour les colonnes suivantes :
    - 'E14TS6', 'E14TS10', 'E14TS20', 'E14TS50', 'E14TS100', 'E14TS200',
       'E14TS500'
- Pour améliorer cela, nous pouvons créer de nouvelles colonnes qui catégorisent les données en fonction de tailles d'entreprises plus vastes :
    - Micro entreprise: 0 <= taille < 10 personnes
    - Petit entreprise: 10 <= taille < 50 personnes
    - Moyenne entreprise :  50 <= taille < 200 personnes
    - Grande Entreprise : taille >= 200 personnes

In [ ]:
summary(df_salary)

In [ ]:
# Descriptive statistics for salary dataset

df_salary.describe()

## Conclusion pour salary DF : 

- Il y a de nombreuses catégories de salariés, mais on peut voir qu'elles sont divisées selon les catégories suivantes : catégorie d'emploi, sexe et âge. Ça peut faciliter analyse. 

In [ ]:
summary(df_population)

## Conclusion pour population DF :
- NIVGEO : une seule valeur unique (COM) est présente
- NB : Il y a un grand nombre de zéros (~74%). Il faut faire attention lors de l'analyse. Peut-être supprimer au future.
- AGEQ80_17 : Il y a un grand nombre de zéros, mais cela semble correct, 0 représente les personnes âgées de 0 à 4 ans. 


## Cleaning, renaming and Merging DFs


### Cleaning df_entreprise : creating new columns, renaming existing columns, changing register

In [ ]:
# Cleaning df_entreprise : creating new columns, renaming existing columns, changing register

df_entreprise.head()


In [ ]:
# Create a new column for the enterprise types

df_entreprise['nb_micro_entreprises'] = df_entreprise['E14TS1'] + df_entreprise['E14TS6']
df_entreprise['nb_small_entreprises'] = df_entreprise['E14TS10'] + df_entreprise['E14TS20']
df_entreprise['nb_medium_entreprises'] = df_entreprise['E14TS50'] + df_entreprise['E14TS100']
df_entreprise['nb_large_entreprises'] = df_entreprise['E14TS200'] + df_entreprise['E14TS500']

In [7]:
def convert_columns_to_lower(df):
    
    """
    Convert column names in the DataFrame to lowercase.
    
    Parameters:
        df (DataFrame): Input DataFrame.
        
    Returns:
        DataFrame: DataFrame with column names converted to lowercase.
    """
    
    df.columns = [col.lower() for col in df.columns]
    return df

In [5]:
# Remove old columns
columns_to_remove = ['E14TS1', 'E14TS6', 'E14TS10', 'E14TS20', 'E14TS50', 'E14TS100', 'E14TS200', 'E14TS500']
df_entreprise = df_entreprise.drop(columns=columns_to_remove)

In [8]:
# Rename the 'E14TST' column to 'total_nb_enterprises'
df_entreprise = df_entreprise.rename(columns={'E14TST': 'total_nb_enterprises', 
                                              'E14TS0ND' : 'nb_unknown_null_size_enterprises'})

# Change the register of the columns 
df_entreprise = convert_columns_to_lower(df_entreprise)

In [ ]:
summary(df_entreprise)

### DataViz df_entreprise

### Cleaning df_name_geographic : creating new columns, renaming existing columns, changing register